forge/forge/op/loss.py

```diff
@@ -119,3 +121,59 @@ def forward(self, predictions, labels):
         square = Multiply("square", diff, diff)
         loss = reduce_loss(self.reduction, square)
         return loss
+
+
+def align_shape(target, reference, name):
+    unsqueezes = 0
+    while target.ndim() < reference.ndim():
+        target = Unsqueeze(f"unsqueeze_{name}_{unsqueezes}", target, unsqueezes)
+        unsqueezes += 1
+    for i in range(target.ndim()):
+        if target.shape[i] != reference.shape[i]:
+            target = Broadcast(f"broadcast_{name}_{i}", target, i, reference.shape[i])
+    return target
+
+
+class BCELoss(ForgeModule):
+    def __init__(self, name: str, reduction: str = "mean"):
+        super().__init__(name)
+        self.reduction = reduction
+        self.is_loss = True
+
+    @validate_shapes(min_dim=1, max_dim=2)
+    def forward(self, prediction, labels):
+        # BCE: -1 * (y * log(p) + (1 - y) * log(1 - p))
+        # First term: y * log(p)
+        log_prediction = Log("log", prediction)
+        first_term = Multiply("mul_lab_pred", labels, log_prediction)
+
+        one = Constant("one", constant=1.0)
+        one = align_shape(one, labels, "one")
```

Comment:
Does Subtract dont support tensor - scalar as it was done in CrossEntropyLoss lines 80-81

Reply:
Subtract isn't the issue, it's the reduce operation afterwards that sometimes has issues with broadcasted operands

I think we should play it safe for now and manually broadcast constants, until that issue is resolved